mono/mini/debugger-agent.c

```diff
@@ -2228,22 +2228,34 @@ static AgentDomainInfo*
 get_agent_domain_info (MonoDomain *domain)
 {
 	AgentDomainInfo *info = NULL;
-
-	mono_domain_lock (domain);
+	MonoJitDomainInfo *jit_info = domain_jit_info (domain);
 
 	info = (AgentDomainInfo *)domain_jit_info (domain)->agent_info;
-	if (!info) {
-		info = g_new0 (AgentDomainInfo, 1);
-		domain_jit_info (domain)->agent_info = info;
-		info->loaded_classes = g_hash_table_new (mono_aligned_addr_hash, NULL);
-		info->source_files = g_hash_table_new (mono_aligned_addr_hash, NULL);
-		info->source_file_to_class = g_hash_table_new (g_str_hash, g_str_equal);
-		info->source_file_to_class_ignorecase = g_hash_table_new (g_str_hash, g_str_equal);
+
+	if (info) {
+		mono_memory_read_barrier ();
```

Comment:
Is this needed even if I'm not dereferencing the pointer?

Reply:
Yes it's needed. Because the caller will access the memory that `info` points to.

Reply:
That's what I figured, but I didn't know if the fact you're coming out of a call means anything in terms of concurrency guarantees.